ocs_ci/ocs/amq.py

```diff
@@ -446,7 +449,194 @@ def run_in_bg(self, namespace=constants.AMQ_NAMESPACE, value='10000', since_time
+        )
+        exec_cmd(cmd=wget_cmd, cwd=self.dir)
+        exec_cmd(cmd=untar_cmd, cwd=self.dir)
+        exec_cmd(cmd=tiller_cmd, cwd=self.dir)
+
+        # Validate tiller pod is running
+        log.info("Waiting for 30s for tiller pod to come up")
+        time.sleep(30)
+        if self.is_amq_pod_running(pod_pattern="tiller", expected_pods=1, namespace=tiller_namespace):
+            log.info("Tiller pod is running")
+        else:
+            raise ResourceWrongStatusException("Tiller pod is not in running state")
+
+        # Create benchmark pods
+        log.info("Create benchmark pods")
+        values = templating.load_yaml(constants.AMQ_BENCHMARK_VALUE_YAML)
+        values["numWorkers"] = num_of_clients
+        benchmark_cmd = (
+            f"linux-amd64/helm install {constants.AMQ_BENCHMARK_POD_YAML}"
+            f" --name {benchmark_pod_name} --tiller-namespace {tiller_namespace}"
+        )
+        exec_cmd(cmd=benchmark_cmd, cwd=self.dir)
+
+        # Making sure the benchmark pod and clients are running
+        if self.is_amq_pod_running(
+            pod_pattern="benchmark", expected_pods=(1 + num_of_clients), namespace=tiller_namespace
+        ):
+            log.info("All benchmark pod is up and running")
+        else:
+            raise ResourceWrongStatusException("Benchmark pod is not getting to running state")
+
+        # Update commonConfig with kafka-bootstrap server details
+        driver_kafka = templating.load_yaml(constants.AMQ_DRIVER_KAFKA_YAML)
+        driver_kafka['commonConfig'] = (
+            f'bootstrap.servers=my-cluster-kafka-bootstrap.{kafka_namespace}.svc.cluster.local:9092'
+        )
+        json_file = f'{self.dir}/driver_kafka'
+        templating.dump_data_to_json(driver_kafka, json_file)
+        cmd = f'cp {json_file} {benchmark_pod_name}-driver:/'
+        self.pod_obj.exec_oc_cmd(cmd)
+
+        # Update the workload yaml
+        if not amq_workload_yaml:
+            amq_workload_yaml = templating.load_yaml(constants.AMQ_WORKLOAD_YAML)
+        yaml_file = f'{self.dir}/amq_workload.yaml'
+        templating.dump_data_to_temp_yaml(amq_workload_yaml, yaml_file)
+        cmd = f'cp {yaml_file} {benchmark_pod_name}-driver:/'
+        self.pod_obj.exec_oc_cmd(cmd)
+
+        # Run the benchmark
+        if worker:
+            cmd = f"bin/benchmark --drivers /driver_kafka --workers {worker} /amq_workload.yaml"
+        else:
+            cmd = "bin/benchmark --drivers /driver_kafka /amq_workload.yaml"
+        log.info(f"Run benchmark and running command {cmd} inside the benchmark pod ")
+        pod_obj = get_pod_obj(name=f"{benchmark_pod_name}-driver", namespace=tiller_namespace)
+        result = pod_obj.exec_cmd_on_pod(command=cmd, out_yaml_format=False, timeout=timeout)
+
+        self.benchmark = True
+
+        # Parse the json output
+        return self.validate_amq_benchmark(benchmark_pod_name, result, amq_workload_yaml)
```

Comment:
Any reason why we are returning the amq benchmark validation in the same run_benchmark function?
What if we want to run some step in the test, while amq_benchmark is in-progress?
Please let me know if I am missing anything here.

Reply:
Sometimes workload wouldn't have run completely or might have failed in between but the test case showed as passed. So added the validation on same function. 

In coming PR will pass `validate_amq_benchmark` as param(bool type).

Reply:
As discussed, removed validating in this function